bin/utest/signal.c

```diff
@@ -295,6 +296,93 @@ TEST_ADD(signal_sigsuspend) {
   return 0;
 }
 
+/* ======= signal_sigtimedwait ======= */
+int sigtimedwait_child(void *arg) {
+  pid_t ppid = getppid();
+  sched_yield();
+  syscall_ok(kill(ppid, SIGUSR1));
+  return 0;
+}
+
+TEST_ADD(signal_sigtimedwait) {
+  syscall_ok(signal(SIGCONT, sigcont_handler));
+  sigset_t set, current, waitset;
+  __sigemptyset(&set);
+  __sigaddset(&set, SIGUSR1);
+  __sigaddset(&set, SIGCONT);
+  syscall_ok(sigprocmask(SIG_SETMASK, &set, NULL));
+
+  utest_spawn(sigtimedwait_child, NULL);
+
+  siginfo_t info;
+  __sigemptyset(&waitset);
+  __sigaddset(&waitset, SIGUSR1);
+  assert(sigtimedwait(&waitset, &info, NULL) == SIGUSR1);
+  assert(info.si_signo == SIGUSR1);
+
+  syscall_ok(sigprocmask(SIG_BLOCK, NULL, &current));
+  assert(__sigsetequal(&set, &current));
+
+  utest_child_exited(0);
+  return 0;
+}
+
+/* ======= signal_sigtimedwait_timeout ======= */
+static int sigusr2_handled = 0;
+
+void sigtimedwait_timeout_sigusr2_handler(int signo) {
```

Comment:
Why don't you name it `sigusr2_handler`?

Reply:
This way I know that some particular test will be the only user of this function. 
Suppose someone in the future would like to use the sigusr2 for testing, then they'd need to create a new function, how would they name it then? They could reuse this function, but I advise against that. If one would like to change the behavior of the handler for some particular test they'd need to create a new function and give it some other name anyway.

Reply:
I suggested it because I see that in other signal tests we name functions that looks like that always in the same way. These functions are used in multiple tests and I think there is nothing wrong in that (they are used identically so why we would like to duplicate code?).

If you would like to add some functionality to such function then it is a reason to make function with unique name (and leave the old one unchanged because we expect some specific behavior from it).

@cahirwpz @pwit81 I think that we can think about making such function a standard. They are used pretty often and maybe we don't have to invent new functions for just handling one signal for every written test?

Reply:
@franciscozdo @korbiniak It's definitely a good idea to factor it out.

I'd do it by extending our current `utest.h` interface by adding:
 * `void signal_count(int signo)` - registers a generic handler for `signo` signal,
 * `int signal_counter(int signo)` - returns a number of times signal `signo` was delivered.

If you have a better idea how to name those functions then feel free to do so.